### PR TITLE
Send hashes instead of real mobile ids via deviceKey <PIX-2449>

### PR DESCRIPTION
**Pull request by grooze**

### CODE REVIEW

File: AppmetrCS/AppMetr.cs [hunk 1]
```diff
@@ -54,9 +54,10 @@ public AppMetr(String url, String token, String mobUuid, String platform, String
 
             _deviceKey = Utils.MakeQueryString(new Dictionary<string, string>
             {
-                {"mobUuid", mobUuid},
-                {"platform", platform},
-                {"mobDeviceType", mobDeviceType}
+                {"token", _token.ToLower()},
+                {"mobUuid", Utils.GetHash(mobUuid)},
+                {"platform", Utils.GetHash(platform)},
```
Comment (chemist777) [inline AppmetrCS/AppMetr.cs]: Should not be cached.

File: AppmetrCS/AppMetr.cs [hunk 2]
```diff
@@ -54,9 +54,10 @@ public AppMetr(String url, String token, String mobUuid, String platform, String
...
+                {"mobDeviceType", Utils.GetHash(mobDeviceType)}
```
Comment (chemist777) [inline AppmetrCS/AppMetr.cs]: Should not be cached.

File: AppmetrCS/Utils.cs [hunk 3]
```diff
@@ -81,5 +82,20 @@ public static String MakeQueryString(Dictionary<String, String> @params)
 
             return queryBuilder.ToString();
         }
+
+        public static string GetHash(String data)
+        {
+            if (String.IsNullOrEmpty(data)) return data;
```
Comment (chemist777) [inline AppmetrCS/Utils.cs]: Null or empty values should not be added to deviceKey.

Comment (grooze) [reply AppmetrCS/Utils.cs]: Null values filtered later in Utils.MakeQueryString

**Pull request merged**